scripts/merge-script/config.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {dirname, resolve} from 'path';
+
+/**
+ * Possible merge methods supported by the Github API.
+ * https://developer.github.com/v3/pulls/#merge-a-pull-request-merge-button.
+ */
+export type GithubApiMergeMethod = 'merge'|'squash'|'rebase';
+
+/**
+ * Target labels represent Github pull requests labels. These labels instruct the merge
+ * script into which branches a given pull request should be merged to.
+ */
+export interface TargetLabel {
+  /** Pattern that matches the given target label. */
+  pattern: RegExp|string;
+  /**
+   * List of branches a pull request with this target label should be merged into.
+   * Can also be wrapped in a function that accepts the target branch specified in the
+   * Github Web UI. This is useful for supporting labels like `target: development-branch`.
+   */
+  branches: string[]|((githubTargetBranch: string) => string[]);
+}
+
+/** Configuration for the merge script. */
+export interface Config {
+  /** Relative path (based on the config location) to the project root. */
+  projectRoot: string;
+  /** Configuration for the upstream repository. */
+  repository: {user: string; name: string; useSsh?: boolean};
+  /** List of target labels. */
+  labels: TargetLabel[];
+  /** Required base commits for given branches. */
+  requiredBaseCommits?: {[branchName: string]: string};
+  /** Pattern that matches labels which imply a signed CLA. */
+  claSignedLabel: string|RegExp;
+  /** Pattern that matches labels which imply a merge ready pull request. */
+  mergeReadyLabel: string|RegExp;
+  /** Label which can be applied to fixup commit messages in the merge script. */
+  commitMessageFixupLabel: string|RegExp;
+  /**
+   * Whether pull requests should be merged using the Github API. This can be enabled
+   * if projects want to have their pull requests show up as `Merged` in the Github UI.
+   * The downside is that fixup or squash commits no longer work as the Github API does
+   * not support this.
+   */
+  githubApiMerge?: {
+    default: GithubApiMergeMethod,
+    labels?: {pattern: string, method: GithubApiMergeMethod}[],
+  };
```

Comment:
Can we require this is defined so its explicit which is being chosen?

```suggestion
  githubApiMerge: false | {
    default: GithubApiMergeMethod,
    labels?: {pattern: string, method: GithubApiMergeMethod}[],
  };
```

Reply:
Yep. Done